src/Messages/Attachments/TSAttachmentStream.m

```diff
@@ -112,30 +116,110 @@ + (NSUInteger)numberOfItemsInAttachmentsFolder
     return count;
 }
 
-- (nullable NSString *)filePath
+- (nullable NSString *)buildLocalFilePath
 {
-    return [MIMETypeUtil filePathForAttachment:self.uniqueId
-                                    ofMIMEType:self.contentType
-                                      filename:self.filename
-                                      inFolder:[[self class] attachmentsFolder]];
+    if (!self.localRelativeFilePath) {
+        return nil;
+    }
+
+    return [[[self class] attachmentsFolder] stringByAppendingPathComponent:self.localRelativeFilePath];
+}
+
+- (nullable NSString *)localFilePathWithTransaction:(YapDatabaseReadWriteTransaction *)transaction
+{
```

Comment:
nit: This method is a little hairy - I imagine because we use it all over the place while the TSAttachment's are in a variety of states.

Also, I'm surprised that this method  might save the attachment. Why does it need to do that?

Would it simplify things to create the localfile path in the corresponding init methods?